lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -682,22 +684,57 @@ def local_user
       ENV['USER']
     end
 
-    # Returns the KeyPair for this host, creating it if needed
+    # Creates the KeyPair for this test run
     #
     # @param region [AWS::EC2::Region] region to create the key pair in
     # @return [AWS::EC2::KeyPair] created key_pair
     # @api private
     def ensure_key_pair(region)
-      @logger.notify("aws-sdk: Ensure key pair exists, create if not")
       key_pairs = region.key_pairs
       pair_name = key_name()
+      delete_key_pair(key_pairs, pair_name, region.name)
```

Comment:
Given the timestamp in the key name, under what circumstances would this not be a noop?